### ISSUE TITLE
Wrong suggestion in `identical conversion` lint

### ISSUE THREAD

**Issue by sigmaSd** [rust-lang/rust-clippy#4183]
Hi,
clippy 0.0.212 (265318d 2019-05-17)

code:  
```rust
fn main() {}

fn _z(a: &std::path::Path) -> std::io::Result<()> {
    for _ in std::fs::read_dir(a)? {}
    Ok(())
}
```
clippy:
```
warning: identical conversion
 --> src/main.rs:4:14
  |
4 |     for _ in std::fs::read_dir(a)? {}
  |              ^^^^^^^^^^^^^^^^^^^^^ help: consider removing `std::fs::read_dir(a)?()`: `std::fs::read_dir(a)?`
  |
  = note: #[warn(clippy::identity_conversion)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#identity_conversion
```
removing `?`  and using `unwrap` for example make the lint warning disappear as it should

**Comment by sigmaSd** [comment]
duplicate https://github.com/rust-lang/rust-clippy/issues/4133
